### ISSUE TITLE
RTL apps with a horizontal scrollView

### ISSUE THREAD

**Issue by hamidhadi** [facebook/react-native#11960]
In the `info.plist`, I changed the `Localization native development region` to 'fa' so my app direction is `RTL`. 
ScrollView horizontal does not work fine. 
I can't see the scroll indicator and can't scrolling.
If I change the `Localization native development region` to `LTR` languages, the scrollview works fine.
Is there anyway to solve this problem?

**Comment by dastoori** [comment]
Same here, Horizontal ScrollView doesn't work properly in RTL Layout (on both iOS and Android)

### Reproduction

Use RTLExample page in UIExplorer to apply RTL layout and then use horizontal ScrollView in `<ScrollView>` page

### Additional Information

* React Native version: 0.40
* Platform: iOS, Android
* Operating System: MacOS

**Comment by dastoori** [comment]
Anyone can help/contribute? 

It's a bug and it breaks my app and a lots of good components like `react-native-swiper` and `react-native-scrollable-tab-view` that uses horizontal ScrollView.

### Example:

This is my code:

```html
<ScrollView horizontal>
  <View style={{ width: 150, height: 100, backgroundColor: '#12a19e' }}><Text>1</Text></View>
  <View style={{ width: 150, height: 100, backgroundColor: '#f45b21' }}><Text>2</Text></View>
  <View style={{ width: 150, height: 100, backgroundColor: '#640fe7' }}><Text>3</Text></View>
  <View style={{ width: 150, height: 100, backgroundColor: '#8f5e3e' }}><Text>4</Text></View>
  <View style={{ width: 150, height: 100, backgroundColor: '#f4bd15' }}><Text>5</Text></View>
  <View style={{ width: 150, height: 100, backgroundColor: '#137cf9' }}><Text>6</Text></View>
</ScrollView>
```

### Preview:

React Native version: 0.41

iOS normal layout:
![ios-ltr](https://cloud.githubusercontent.com/assets/2003143/22634763/a8374e38-ec42-11e6-99a3-f37532ac9027.gif)

iOS RTL layout:
![ios-rtl](https://cloud.githubusercontent.com/assets/2003143/22634771/b2143254-ec42-11e6-8001-6e02844a59e3.gif)

Android normal layout:
![android-ltr](https://cloud.githubusercontent.com/assets/2003143/22634772/bdb85a22-ec42-11e6-974e-9d3d90065761.gif)

Android RTL layout
![android-rtl mov](https://cloud.githubusercontent.com/assets/2003143/22634780/ce7a9884-ec42-11e6-9f3a-9e94bad3adba.gif)

### Related issue:

https://github.com/skv-headless/react-native-scrollable-tab-view/issues/508

**Comment by dastoori** [comment]
Temporary fix:

```javascript
<ScrollView
  horizontal
  style={{ flexDirection: I18nManager.isRTL ? 'row-reverse' : 'row' }}
>
   ...
</ScrollView>
```

In RTL layout you should use [`scrollToEnd()` method](https://facebook.github.io/react-native/docs/scrollview.html#scrolltoend) to scroll to the first item

**Comment by TGNC** [comment]
This is a deal breaker, someone is on it?

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while.

But don't worry, it will live on with Canny! Check out its new home: https://react-native.canny.io/feature-requests/p/rtl-apps-with-a-horizontal-scrollview

**Comment by malashkevich** [comment]
Hey, this is still a problem, guys. :(
Can we reopen this issue?

**Comment by DanGDroid** [comment]
tried updating rn up to 49 still horizontal flatlist on rtl does not function

**Comment by dastoori** [comment]
It seems that it's fixed on [v0.51.0](https://github.com/facebook/react-native/releases/tag/v0.51.0) ([this commit](https://github.com/facebook/react-native/commit/c278020633119b0c20752593666b0e8536f6a4b1))
Can anyone test this?

**Comment by hramos** [comment]
I won't be reopening this issue as it does not contain sufficient information as required by the template. You may open a new one.
